addons/t9n/models/resource.py

```diff
@@ -1,11 +1,36 @@
-from odoo import fields, models
+from odoo import fields, models, api, _, Command
+from odoo.exceptions import ValidationError
+import polib
+import base64
+
+
+def decode_resource_file(resource_file):
+    try:
+        file_content = base64.b64decode(resource_file).decode("utf-8")
+        po_obj = polib.pofile(file_content)
+    except:
+        raise ValidationError(_("The file should be a .po file"))
```

Comment:
Do you know the type of exception that will be thrown here if the file is not a .po file? It's good practice to always provide the specific type of exception that is expected. If you're not specific enough, your code could end up catching irrelevant things, such as keyboard interrupts (interrupting the program with Ctrl + D)